api/v1alpha2/conversion.go

```diff
@@ -202,6 +204,7 @@ func (src *MachineDeployment) ConvertTo(dstRaw conversion.Hub) error {
 	// This conversion can be overwritten when restoring the ClusterName field.
 	if name, ok := src.Labels[MachineClusterLabelName]; ok {
 		dst.Spec.ClusterName = name
+		dst.Spec.Template.Spec.ClusterName = name
```

Comment:
This conversion will result in a changed hash for the deployment. We'll need to keep that in mind with the v1alpha3 release notes.

Reply:
xref https://kubernetes.slack.com/archives/C8TSNPY4T/p1580231398178200 - conversation about whether or not a different hash induces a rolling upgrade.

also xref #1938 where we previously discussed this concern

Reply:
Recording here for posterity: we want to do #2183, and the hash is not relevant in terms of rolling updates.